pkg/network/handlers/work_report.go

```diff
@@ -0,0 +1,74 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/eigerco/strawberry/internal/block"
+	"github.com/eigerco/strawberry/internal/crypto"
+	"github.com/eigerco/strawberry/internal/refine"
+	"github.com/eigerco/strawberry/internal/service"
+	"github.com/eigerco/strawberry/internal/work"
+)
+
+// ProduceWorkReport runs the refine invocation and returns the work report along with the wp hash
+func ProduceWorkReport(
+	ctx context.Context,
+	refine refine.RefinePVMInvoker,
+	serviceState service.ServiceState,
+	authOutput []byte,
+	coreIndex uint16,
+	bundle work.PackageBundle,
+	segmentRootLookup map[crypto.Hash]crypto.Hash,
+) (*block.WorkReport, *crypto.Hash, error) {
```

Comment:
Is this a pointer because the work report might be nil after producing?

Also given that the work report has a `Hash()` method, is there any advantage to returning both the work report and it's hash here, as opposed to just the work report?

Reply:
I return nil when there's an error happy to drop the pointer if you wish.
I removed the hash from the return values

Reply:
Up to you, but a non pointer is probably a bit cleaner. :)